ThemeType breakpoints missing types for medium and large?

When I try to set `edgeSize` for breakpoints `medium` and `large` my TypeScript compiler complain about `edgeSize` not being a valid type on `global.breakpoints.medium` and `global.breakpoints.large`.

### Expected Behavior

Typescript builds the project.

### Actual Behavior

Typescript throws the following type error:

```
src/theme/index.ts:28:9 - error TS2322: Type '{ edgeSize: typeof edgeSize; }' is not assignable to type '{ value?: number; }'.
  Object literal may only specify known properties, and 'edgeSize' does not exist in type '{ value?: number; }'.

28         edgeSize,
```

### Steps to Reproduce

1. Add the following config to the theme:

```
global: {
  breakpoints: {
    medium: {
      edgeSize: {
        small: '1px'
      }
    }
  }
}
```

### Your Environment

- Grommet version: 2.11.0
- TypeScript

--- a/src/js/components/Grid/stories/Responsive.js
+++ b/src/js/components/Grid/stories/Responsive.js
@@ -20,7 +20,9 @@ const customBreakpoints = deepMerge(grommet, {
       medium: {
         value: 900,
       },
-      large: 3000,
+      large: {
+        value: 3000,
+      },
     },
   },
 });

--- a/src/js/components/Menu/stories/typescript/Themed.tsx
+++ b/src/js/components/Menu/stories/typescript/Themed.tsx
@@ -1,0 +1,83 @@
+import * as React from 'react';
+import { storiesOf } from '@storybook/react';
+import isChromatic from 'storybook-chromatic/isChromatic';
+
+import { Grommet, Box, Menu, ThemeType } from 'grommet';
+import { Next } from 'grommet-icons';
+
+const customBreakpoints: ThemeType = {
+  global: {
+    breakpoints: {
+      small: {
+        value: 600,
+        edgeSize: {
+          small: '1px',
+        },
+        borderSize: {
+          small: '2px',
+        },
+        size: {
+          xxsmall: '24px',
+        },
+      },
+      medium: {
+        value: 900,
+        borderSize: {
+          small: '2px',
+        },
+        edgeSize: {
+          small: '1px',
+        },
+        size: {
+          xxsmall: '24px',
+        },
+      },
+      large: {
+        value: 3000,
+        borderSize: {
+          small: '2px',
+        },
+        edgeSize: {
+          small: '1px',
+        },
+        size: {
+          xxsmall: '24px',
+        },
+      },
+      xlarge: {
+        value: 3000,
+        borderSize: {
+          small: '2px',
+        },
+        edgeSize: {
+          small: '1px',
+        },
+        size: {
+          xxsmall: '24px',
+        },
+      },
+    },
+  },
+};
+const App = () => {
+  return (
+    <Grommet theme={customBreakpoints}>
+      <Box align="center" pad="large">
+        <Menu
+          dropProps={{ align: { top: 'bottom', left: 'left' } }}
+          label="actions"
+          icon={<Next />}
+          items={[
+            { label: 'Launch', onClick: () => {} },
+            { label: 'Abort', onClick: () => {} },
+            { label: 'Disabled', disabled: true },
+          ]}
+        />
+      </Box>
+    </Grommet>
+  );
+};
+
+if (!isChromatic()) {
+  storiesOf('TypeScript/Menu', module).add('Themed', () => <App />);
+}

--- a/src/js/themes/base.d.ts
+++ b/src/js/themes/base.d.ts
@@ -1,6 +1,9 @@
 import { 
   BackgroundType, 
   BorderType,
+  BreakpointBorderSize,
+  BreakpointEdgeSize,
+  BreakpointSize,
   ColorType,  
   DeepReadonly, 
   GapType,
@@ -96,38 +99,27 @@ export interface ThemeType {
     breakpoints?: {
       small?: {
         value?: number;
-        borderSize?: {
-          xsmall?: string;
-          small?: string;
-          medium?: string;
-          large?: string;
-          xlarge?: string;
-        };
-        edgeSize?: {
-          none?: string;
-          hair?: string;
-          xxsmall?: string;
-          xsmall?: string;
-          small?: string;
-          medium?: string;
-          large?: string;
-          xlarge?: string;
-        };
-        size?: {
-          xxsmall?: string;
-          xsmall?: string;
-          small?: string;
-          medium?: string;
-          large?: string;
-          xlarge?: string;
-          full?: string;
-        };
+        borderSize?: BreakpointBorderSize;
+        edgeSize?: BreakpointEdgeSize;
+        size?: BreakpointSize;
       };
       medium?: {
         value?: number;
+        borderSize?: BreakpointBorderSize;
+        edgeSize?: BreakpointEdgeSize;
+        size?: BreakpointSize;
       };
       large?: {
         value?: number;
+        borderSize?: BreakpointBorderSize;
+        edgeSize?: BreakpointEdgeSize;
+        size?: BreakpointSize;
+      };
+      [x: string]: {    
+        value?: number;
+        borderSize?: BreakpointBorderSize;
+        edgeSize?: BreakpointEdgeSize;
+        size?: BreakpointSize;
       };
     };
     deviceBreakpoints?: {

--- a/src/js/utils/index.d.ts
+++ b/src/js/utils/index.d.ts
@@ -62,3 +62,35 @@ export type PadType = EdgeType;
 export type PlaceHolderType = string | JSX.Element | React.ReactNode;
 export type RoundType = boolean | "xsmall" | "small" | "medium" | "large" | "xlarge" | "full" | string | {corner?: "top" | "left" | "bottom" | "right" | "top-left" | "top-right" | "bottom-left" | "bottom-right",size?: "xsmall" | "small" | "medium" | "large" | "xlarge" | string};
 export type TextAlignType = "start" | "center" | "end";
+
+declare const breakpointEdgeSize: {
+  none?: string;
+  hair?: string;
+  xxsmall?: string;
+  xsmall?: string;
+  small?: string;
+  medium?: string;
+  large?: string;
+  xlarge?: string;
+};
+export type BreakpointEdgeSize = typeof breakpointEdgeSize;
+
+declare const breakpointBorderSize: {
+    xsmall?: string;
+    small?: string;
+    medium?: string;
+    large?: string;
+    xlarge?: string;
+}
+export type BreakpointBorderSize = typeof breakpointBorderSize;
+
+declare const breakpointSize: {          
+  xxsmall?: string;
+  xsmall?: string;
+  small?: string;
+  medium?: string;
+  large?: string;
+  xlarge?: string;
+  full?: string;
+};
+export type BreakpointSize = typeof breakpointSize;

